mllib/src/main/scala/org/apache/spark/mllib/regression/IsotonicRegression.scala

```diff
@@ -408,8 +409,12 @@ class IsotonicRegression private (private var isotonic: Boolean) extends Seriali
    */
   private def parallelPoolAdjacentViolators(
       input: RDD[(Double, Double, Double)]): Array[(Double, Double, Double)] = {
-    val parallelStepResult = input
-      .sortBy(x => (x._2, x._1))
+    val keyedInput = input
```

Comment:
`repartitionAndSortWithinPartitions` requires the partition key and the sortBy key to be the same. We want to partition by feature, then sort by feature *and* label. So it would still require a second step to sort the partitions, although they'd be mostly sorted already. Maybe there could be a speed-up by using an insertion or Timsort on the mostly-sorted partition...